### PR TITLE
Don't write zeros when encoding baseX numbers

### PR DESCRIPTION
**Pull request by steveluscher**
# Test Plan

No change in benchmarks.

```shell
cd packages/codecs-strings/
pnpm benchmark
```

### CODE REVIEW

**Comment by steveluscher** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2142?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `master`
  * **PR #2139** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2139?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #2140** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2140?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #2141** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2141?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #2142** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2142?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2142?utm_source=stack-comment).

File: packages/codecs-strings/src/baseX.ts [hunk 1]
```diff
@@ -31,7 +31,6 @@ export const getBaseXEncoder = (alphabet: string): VariableSizeEncoder<string> =
             // Handle leading zeroes.
             const [leadingZeroes, tailChars] = partitionLeadingZeroes(value, alphabet[0]);
             if (!tailChars) {
-                bytes.set(new Uint8Array(leadingZeroes.length).fill(0), offset);
```
Comment (steveluscher) [inline packages/codecs-strings/src/baseX.ts:34]: This array is already preallocated with zeros, so I think there's no need to write zeros, unless I'm mistaken.

Comment (lorisleiva) [reply packages/codecs-strings/src/baseX.ts:34]: Hmm that's not always true. Someone can use the `write` method of the Encoder using a custom buffer. When they do, they expect the bytes at the given offset to be overwritten with the encoded string bytes. Perhaps we should add a regression test for this.

**Review by lorisleiva — CHANGES_REQUESTED** [review]
See comment on other thread. Also, wen benchmark changes? 😅

**Pull request closed without merging**